library/kani/src/futures.rs

```diff
@@ -41,3 +44,216 @@ const NOOP_RAW_WAKER: RawWaker = {
 
     RawWaker::new(std::ptr::null(), &RawWakerVTable::new(clone_waker, noop, noop, noop))
 };
+
+static mut EXECUTOR: Scheduler = Scheduler::new();
+const MAX_TASKS: usize = 16;
+
+type BoxFuture = Pin<Box<dyn Future<Output = ()> + Sync + 'static>>;
+
+/// Allows to parameterize how the scheduler picks the next task to poll in `spawnable_block_on`
+pub trait SchedulingStrategy {
+    /// Picks the next task to be scheduled whenever the scheduler needs to pick a task to run next, and whether it can be assumed that the picked task is still running
+    ///
+    /// Tasks are numbered `0..num_tasks`.
+    /// For example, if pick_task(4) returns (2, true) than it picked the task with index 2 and allows Kani to `assume` that this task is still running.
+    /// This is useful if the task is chosen nondeterministicall (`kani::any()`) and allows the verifier to discard useless execution branches (such as polling a completed task again).
+    fn pick_task(&mut self, num_tasks: usize) -> (usize, bool);
+}
+
+impl<F: FnMut(usize) -> usize> SchedulingStrategy for F {
+    #[inline]
+    fn pick_task(&mut self, num_tasks: usize) -> (usize, bool) {
+        (self(num_tasks), false)
+    }
+}
+
+/// Keeps cycling through the tasks in a deterministic order
+#[derive(Default)]
+pub struct RoundRobin {
+    index: usize,
+}
+
+impl SchedulingStrategy for RoundRobin {
+    #[inline]
+    fn pick_task(&mut self, num_tasks: usize) -> (usize, bool) {
+        self.index = (self.index + 1) % num_tasks;
+        (self.index, false)
+    }
+}
+
+/// Picks the next task nondeterministically
+#[derive(Default)]
+pub struct NondeterministicScheduling;
+
+impl SchedulingStrategy for NondeterministicScheduling {
+    #[cfg(kani)]
```

Comment:
I don't think there is any need for this. We only distribute the crate with `kani` enabled. Additionally, `crate::any()` should fail already if this is used outside of Kani.

Reply:
OK, I'll remove the `#[cfg]`s